driver-sync/src/test/functional/com/mongodb/client/unified/ContextElement.java

```diff
@@ -467,4 +508,40 @@ private static BsonDocument serverMonitorEventToDocument(final Object event) {
         return new BsonDocument(EventMatcher.getEventType(event.getClass()),
                 new BsonDocument("awaited", BsonBoolean.valueOf(EventMatcher.getAwaitedFromServerMonitorEvent(event))));
     }
+
+    static BsonDocument topologyEventToDocument(final Object event) {
+        if (event != null && !(event instanceof ClusterOpeningEvent || event instanceof ClusterDescriptionChangedEvent || event instanceof ClusterClosedEvent)) {
+            throw new UnsupportedOperationException("Unsupported topology event: " + event.getClass().getName());
+        }
+        BsonDocument eventDocument = new BsonDocument();
+        if (event instanceof ClusterDescriptionChangedEvent) {
+            ClusterDescriptionChangedEvent changedEvent = (ClusterDescriptionChangedEvent) event;
+            eventDocument.put("previousDescription",
+                    new BsonDocument("type", new BsonString(clusterDescriptionToString(changedEvent.getPreviousDescription()))));
+            eventDocument.put("newDescription",
+                    new BsonDocument("type", new BsonString(clusterDescriptionToString(changedEvent.getNewDescription()))));
+        }
+        return new BsonDocument(EventMatcher.getEventType(event.getClass()), eventDocument);
+    }
+
+    static String clusterDescriptionToString(final ClusterDescription clusterDescription) {
+        switch (clusterDescription.getType()) {
+            case STANDALONE:
+                return "Single";
+            case REPLICA_SET:
+                return clusterDescription.getServerDescriptions().stream().
+                        filter(ServerDescription::isPrimary)
+                        .findFirst()
+                        .map(serverDescription -> "ReplicaSetWithPrimary")
+                        .orElse("ReplicaSetNoPrimary");
+            case SHARDED:
+                return "Sharded";
+            case LOAD_BALANCED:
+                return "LoadBalancer";
+            case UNKNOWN:
+                return "Unknown";
+            default:
+                throw new IllegalStateException("Unexpected value: " + clusterDescription.getShortDescription());
```

Comment:
In existing code pattern, it seems either `UnsupportedOperationException` (by Jeff) or `Assertions.fail()` (by Valentin) for such enum slipping through default branch.

Reply:
Opted for `UnsupportedOperationException` as that appears to be the norm here for ContextElements.